### PR TITLE
ES6: Convert discussion/discussion-frontend

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `discussion/discussion-frontend` to ES6.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import config from 'lib/config';
+import fastdom from 'fastdom';
+import { integerCommas } from 'lib/formatters';
+import { loadScript } from 'lib/load-script';
+import mediator from 'lib/mediator';
+import reportError from 'lib/report-error';
+import { Loader } from './loader';
+
+const loadDiscussionFrontend = (loader: Loader, opts: Object) => {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js]: Missing a return type here 😄

File: static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        const formatted = integerCommas(value);
+
+        if (formatted) {
+            fastdom.write(() => {
+                element.textContent = formatted;
+            });
+        }
+    };
+
+    const onDiscussionFrontendLoad = (emitter): void => {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js]: What is `emitter`'s type?

File: static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        emitter.on('error', (feature, error) => {
+            reportError(error, { feature: `discussion-${feature}` }, false);
+        });
+
+        emitter.once('comment-count', (value: string): void => {
+            if (value === 0) {
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js]: if `value` is a string type `value === 0` will never be true?

File: static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                loader.setState('empty');
+            } else {
+                /* By the time discussion frontent loads, the number of comments
+                   might have changed. If there are other comment counts element
+                   in the page refresh their value. */
+                const otherValues = document.getElementsByClassName(
+                    'js_commentcount_actualvalue'
+                );
+
+                if (otherValues.length) {
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js]: I don't think you need the `otherValues.length` check here, `[...otherValues]` will equal `[]`  if `otherValues.length` is 0, so the forEach won't run.

File: static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js (hunk 3 above)
Comment (gustavpursche) [reply static/src/javascripts/projects/common/modules/discussion/discussion-frontend.js]: @SiAdcock [had an idea, where this code comes from - `discussion-frontend`](https://github.com/guardian/discussion-frontend/blob/5d193d6fb3e344eb75891252fc9dba3acee4c969/src/model/discussion.js#L25-L31).

**Comment by gustavpursche** [comment]
@SiAdcock @GHaberis Fixed all the things you mentioned. Ready for another 👀

**Review by sndrs — APPROVED** [review]
lgtm

**Pull request merged**